CMakeLists.txt

```diff
@@ -34,51 +34,33 @@
 # policies, either expressed or implied, of any organization.
 # #L%
 
-cmake_minimum_required(VERSION 2.8.12)
-
-# Default policy is from 2.8.12
-cmake_policy(VERSION 2.8.12)
-# Set MacOSX @rpath usage globally.
-if (POLICY CMP0020)
-  cmake_policy(SET CMP0020 NEW)
-endif(POLICY CMP0020)
-if (POLICY CMP0042)
-  cmake_policy(SET CMP0042 NEW)
-endif(POLICY CMP0042)
-# Use new variable expansion policy.
-if (POLICY CMP0053)
-  cmake_policy(SET CMP0053 NEW)
-endif(POLICY CMP0053)
-if (POLICY CMP0054)
-  cmake_policy(SET CMP0054 NEW)
-endif(POLICY CMP0054)
-
-# For MSVC
-enable_language(CXX)
-# For feature tests with CMake 2.8.x
-enable_language(C)
+cmake_minimum_required(VERSION 3.2.0)
+cmake_policy(VERSION 3.2.0)
 
-list(APPEND CMAKE_MODULE_PATH
-     "${CMAKE_CURRENT_LIST_DIR}/cpp/cmake")
+# For MSVC and FindThreads, respectively.
+enable_language(CXX C)
 
-include("${CMAKE_CURRENT_LIST_DIR}/cpp/cmake/BioFormatsCommon.cmake")
+list(APPEND CMAKE_MODULE_PATH
+     "${CMAKE_CURRENT_LIST_DIR}/cmake")
 
-project(bioformats)
-message(STATUS "Configuring Bio-Formats main build")
+include(cmake/Version.cmake)
+ome_project_version("OME Common C++" "${CMAKE_CURRENT_SOURCE_DIR}")
 
-include(cpp/cmake/Version.cmake)
+project(ome-common
+        VERSION "${OME_VERSION_MAJOR}.${OME_VERSION_MINOR}.${OME_VERSION_PATCH}"
+        LANGUAGES CXX)
```

Comment:
This comes from the git tag when building from git, or the embedded version file when building from a source release.  At the present, this is 5.2.0, set from the v5.2.0-m1 tag.